### DISCUSSION TITLE
How do i host streamlit projects ?

### DISCUSSION THREAD

**Discussion by DarshanAnand007** [community/community#134286] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

How do i host streamlit projects fro free
my repo : https://github.com/DarshanAnand007/SEN-DSUU.git

**Comment by LiteBrite82** [comment]
@DarshanAnand007 
If it is a front end only application, you could use GitHub Pages

**Comment by DarshanAnand007** [comment]
I have ollama embedead with it ,can u help me host ?

**Reply by LiteBrite82** [reply]
your repo is private (I get a 404 when I click the link you provided). 

It could be done, but security would be an issue. You would have to connect your local machine through the deployed front end; I'm not aware of a way to hide API keys or Auth in GitHub Pages.

**Reply by DarshanAnand007** [reply]
please recheck I have made it public , and thank you for showing up intrest

**Reply by LiteBrite82** [reply]
oh snap! you included your .env file in the repo - it should not be there. .env's should only be on your local machine. Create a `.gitignore` file and add ".env" to line 1. 

It would be good to cancel the keys that are currently in the repo and generate new ones for security purposes.

**Reply by LiteBrite82** [reply]
be sure to remove the .env file from the repo entirely

**Comment by LiteBrite82** [comment, marked as answer]
As far as hosting goes, you could potentially create a PWA as a user interface, but you would have to host the backend on a different provider. if you are controlling the API (custom built by you) you would have to create and authorization (log in) method between the front end and back end to avoid exposing your API keys. 

It would be easier to use a host service that supports both front end and backend so you can use the .env files to secure the API keys.

Ive used Heroku and Netlify in the past for my school projects.

**Reply by LiteBrite82** [reply]
but thats all from the perspective of a web based app
